libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -123,15 +123,29 @@ void DocStringAnalyser::parseDocStrings(
 )
 {
 	DocStringParser parser;
-	if (_node.documentation() && !_node.documentation()->empty())
+	if (_node.documentation() && !_node.documentation()->text())
 	{
-		if (!parser.parse(*_node.documentation(), m_errorReporter))
+		if (!parser.parse(*_node.documentation()->text(), m_errorReporter))
 			m_errorOccured = true;
 		_annotation.docTags = parser.tags();
 	}
 	for (auto const& docTag: _annotation.docTags)
 		if (!_validTags.count(docTag.first))
 			appendError("Doc tag @" + docTag.first + " not valid for " + _nodeName + ".");
+		else
+			if (docTag.first == "return")
+				if (auto* function = dynamic_cast<FunctionDefinition const*>(&_node))
+				{
+					bool foundMatch = false;
+					for (auto const& returnParameter: function->returnParameters())
+					{
+						auto firstWord = docTag.second.content.substr(docTag.second.content.find_first_of(" \t"));
+						if (firstWord == returnParameter->name())
+							foundMatch = true;
+					}
+					if (!foundMatch)
+						appendError("Doc tag @" + docTag.first + " does not match named return parameter.");
```

Comment:
Do we call it "Doc tag" in other places?

Reply:
That particular change is not part of this PR any more. Moved it to https://github.com/ethereum/solidity/pull/7837.
But yes, we do use that wording. Please see the other error reported in the very same function.